src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -250,11 +250,27 @@ public double getDistanceMeters() {
     @Override
     public State traverse(State s0) {
         final RoutingRequest options = s0.getOptions();
-        StateEditor editor = doTraverse(s0, options, s0.getNonTransitMode());
-        State state = (editor == null) ? null : editor.makeState();
+        final StateEditor editor;
+
```

Comment:
What is the difference between what happens here and the transitions that happen at line 390?

Reply:
The logic here determines how to traverse _this_ edge, while the transitions around line 390 determine if an actual cost needs to be applied for the switching. I've moved the transition / cost checking to `BikeWalkableEge`.